fcrepo-exts/fcrepo-migration-validator
# Add validation for BINARY_SIZE on managed datastreams

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3687

# What does this Pull Request do?

This adds a comparison of the file sizes as read from the filesystem for managed datastreams. 

# What's new?

* Added validation for comparing file sizes between F3 and F6 repositories
* Switch to the maven-shade-plugin for creating the runnable jar 

# How should this be tested?

Run the validator against your data set and check that BINARY_SIZE validations show up in the results.

e.g.
```
java -jar target/fcrepo-migration-validator-0.1.0-SNAPSHOT-driver.jar -s akubra -d src/test/resources/test-object-validation/deleted-validation-it/deleted-object/f3/datastreams -o src/test/resources/test-object-validation/deleted-validation-it/deleted-object/f3/objects -c src/test/resources/test-object-validation/deleted-validation-it/deleted-object/f6/data/ocfl-root -r validation-output
```

# Additional Notes:

The dependency for `migration-utils` needed to be updated in order to get changes to `DatastreamVersion` which allows us to get access to the File in a F3 repository. This update also brought in an updated version of apache-jena which caused the validator to thrtow NullPointerExceptions when trying to read the n-triples. Looking at the [Jena documentation](https://jena.apache.org/documentation/notes/jena-repack.html), this was likely a result of Jena not initializing correctly due to the assembly plugin not shading its resources. I opted to move packaging of this project to the maven-shade-plugin because of this, and kept everything as in line as I could with what was defined in the maven-assembly-plugin.

# Interested parties

@dbernstein
